bin/rad-patch

```diff
@@ -375,17 +360,25 @@
 
 (def args (get-args!))
 
+(def patch-num-pat (/number-string (/integer 'patch-number)))
+
 (machine/catch-daemon!
  (fn []
    (match args
           (/list-cmd 'options) (list-items (machine-name) options)
           (/cmd-1 "propose" 'commit help) (new-patch! (machine-name) commit)
-          (/cmd-1 "checkout" 'patch-number help) (checkout-patch! (machine-name) (read patch-number))
-          (/cmd-1 "show" 'patch-number help) (show-patch! (machine-name) (read patch-number))
-          (/cmd-1 "retract" 'patch-number help) (edit-patch-state! (machine-name) (read patch-number) :retracted)
-          (/cmd-1 "accept" 'patch-number help) (edit-patch-state! (machine-name) (read patch-number) :accepted)
-          (/cmd-1 "reject" 'patch-number help) (edit-patch-state! (machine-name) (read patch-number) :rejected)
-          (/cmd-2 "comment" 'patch-number 'comment help) (create-comment! (machine-name) (read patch-number) comment)
+          (/cmd-1 "checkout" patch-num-pat help) (checkout-patch! (machine-name) patch-number)
+          (/cmd-1 "checkout" _ help)             (cmd-parse-failure (error/whole-item-number :patch))
+          (/cmd-1 "show" patch-num-pat help) (show-patch! (machine-name) patch-number)
+          (/cmd-1 "show" _ help)                (cmd-parse-failure (error/whole-item-number :patch))
+          (/cmd-1 "retract" patch-num-pat help) (edit-patch-state! (machine-name) patch-number :retracted)
+          (/cmd-1 "retract" patch-num-pat help) (cmd-parse-failure (error/whole-item-number :patch))
+          (/cmd-1 "accept" patch-num-pat help) (edit-patch-state! (machine-name) patch-number :accepted)
+          (/cmd-1 "accept" _ help)                (cmd-parse-failure (error/whole-item-number :patch))
+          (/cmd-1 "reject" patch-num-pat help) (edit-patch-state! (machine-name) patch-number :rejected)
+          (/cmd-1 "reject" _ help)                (cmd-parse-failure (error/whole-item-number :patch))
+          (/cmd-2 "comment" patch-num-pat 'comment help) (create-comment! (machine-name) patch-number comment)
+          (/cmd-2 "comment" _ 'comment help) (cmd-parse-failure (error/whole-item-number :patch))
```

Comment:
It's more and more difficult to keep an overview of this. Could the `patch-num-pat` or `_` commands somehow be extracted into one helper command?

Reply:
Let me try.